tests/Unit/Domain/FunctionsOfTime/Test_PiecewisePolynomial.cpp

```diff
@@ -194,52 +195,9 @@ void test_serialization_versioning() {
                              5.0));
   poly->update(5.0, {6.0}, 7.0);
 
-  // Because of the implementation-defined sign, there's no way to
-  // write char literals that won't cause narrowing errors.
-  const auto vector_char = [](auto... values) {
-    return std::vector<char>{static_cast<char>(values)...};
-  };
-
-  // After any serialization change, generate a new set of bytes with:
-  // for (char c : serialize(poly)) {
-  //   printf("0x%02hhx, ", c);
-  // }
-  const auto serialization_v2 = vector_char(
-      0x1d, 0x3c, 0x4f, 0x83, 0x49, 0x40, 0x19, 0xac, 0x02, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1c, 0x40,
-      0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x40, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x40, 0x01, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10, 0x40,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40, 0x01, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x2e, 0x40,
-      0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x18, 0x40);
-  const auto serialization_v3 = vector_char(
-      0x1d, 0x3c, 0x4f, 0x83, 0x49, 0x40, 0x19, 0xac, 0x03, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x1c, 0x40, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x2e, 0x40, 0x01, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x18,
-      0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14, 0x40, 0x01, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x08, 0x40, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x10, 0x40, 0x01);
-  REQUIRE(serialize(poly) == serialization_v3);
-
-  {
-    std::unique_ptr<FunctionsOfTime::FunctionOfTime> deserialized{};
-    deserialize(make_not_null(&deserialized), serialization_v2.data());
-    CHECK(dynamic_cast<const Poly&>(*poly) ==
-          dynamic_cast<const Poly&>(*deserialized));
-  }
-  {
-    std::unique_ptr<FunctionsOfTime::FunctionOfTime> deserialized{};
-    deserialize(make_not_null(&deserialized), serialization_v3.data());
-    CHECK(dynamic_cast<const Poly&>(*poly) ==
-          dynamic_cast<const Poly&>(*deserialized));
-  }
+  TestHelpers::serialization::test_versioning<Poly>(
+      "Domain/FunctionsOfTime/PiecewisePolynomial.serializations", "version 3",
```

Comment:
Can you keep the test for v2 around as well?

Reply:
?  It's there.